### PR TITLE
Replace `.click()`, `.focus()` and `.attr()` jQuery functions with equivalent native JS

### LINKED ISSUE #919: Convert jQuery `.click()`, `.focus()`, `.attr()` and other homologous functions to their native equivalents
Sub-issue of #367.

### PR DESCRIPTION
**Pull request by dheerajdlalwani**
I have tested the changes in Edge Chromium, Chrome, Firefox and IE 11.

So far everything has been working. Opening PR. Let me know if any changes are required :)

Fixes #919 .

### CODE REVIEW

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:357]: It's minor, but could you please use the same quotation style as the surrounding code? I.e. in this case, use single quotation marks instead of double.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:433]: It looks like there's an opportunity to regularize the quotation mark style here -- could you change all the above to single quotation marks?

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:434]: Remove unneeded spare line above.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1095]: Use single quotation marks.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1298]: Let's regularize here as well -- use single quotes.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1368]: Could you convert this to native as well? It shouldn't be difficult.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1370]: Use single quotation marks.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1398]: Opportunity to regularize - use single quotation marks.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1397]: For the sake of consistency, change the double quotation marks in this systemAlert to single. I realize this wasn't part of the PR, but in this case, there's an opportunity to regularize the code in this area.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:2116]: Use single quotation marks

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:386]: I suggest doing this .val() statement as well (i.e. convert to native).

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:389]: Perhaps you can do these two empty() statements as well. See https://youmightnotneedjquery.com/?support=ie11#empty .

**Review by Jaifroid — CHANGES_REQUESTED** [review]
Thank you very much for this PR. It's looking good! I haven't tested yet -- I've got some minor changes, mostly to do with quotation mark style, and a few cases where you could convert some other functions as well which don't look difficult to do. I'll test once you've made changes.

**Comment by dheerajdlalwani** [comment]
> Thank you very much for this PR. It's looking good! I haven't tested yet -- I've got some minor changes, mostly to do with quotation mark style, and a few cases where you could convert some other functions as well which don't look difficult to do. I'll test once you've made changes.

Sure! I'll make the changes.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:394]: This isn't quite right. The jQuery function `contents()` gets all the child nodes, I think, of the iframe (in this case). You'll need to iterate over them and empty them all. If you don't feel confident about this, you can revert this back to the jQuery function, and we can deal with it in a different issue/PR.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:1070]: Single quotation marks here too, please.

**Review by Jaifroid — CHANGES_REQUESTED** [review]
Nearly there -- just a few more small things!

File: www/js/app.js
Comment (Jaifroid) [reply www/js/app.js:394]: Thinking about it, it's possible your code has the same effect as the original jQuery statement, as it would iterate over every child node anyway. It needs to be checked that we end up with a completely empty/purged iframe.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:409]: Remove extraneous empty line.

**Review by Jaifroid — APPROVED** [review]
This looks good, thanks (there is just one minor thing, removing a blank line). I'll do some testing later today, in particular on the effect of purging the iframe.

**Comment by dheerajdlalwani** [comment]
> This looks good, thanks (there is just one minor thing, removing a blank line). I'll do some testing later today, in particular on the effect of purging the iframe.

Great. I've removed the empty line. And sure, do test and let me know if there are any issues. I'll be most happy to resolve them :)

File: www/js/app.js
Comment (Jaifroid) [reply www/js/app.js:394]: OK, I've determined this code doesn't ever run, because articleContent is the iframe, and the iframe doesn't have any child elements. We need to select the content document. I'll make a suggestion.

File: www/js/app.js
Comment (dheerajdlalwani) [reply www/js/app.js]: Removed :)
Amended the previous commit and force pushed.

File: www/js/app.js
Comment (Jaifroid) [inline www/js/app.js:394]:
```suggestion
        let articleContent = document.getElementById('articleContent');
        let articleContentDoc = articleContent ? articleContent.contentDocument : null;
        while (articleContentDoc.firstChild) articleContentDoc.removeChild(articleContentDoc.firstChild);
```
Here is my suggestion. It needs testing.

File: www/js/app.js
Comment (Jaifroid) [reply www/js/app.js]: OK, but JFYI it's generally best not to force push while you're in a PR others are reviewing, because it forces them to reset their copy. It doesn't matter if the commit history gets long, because you'll squash to a single commit before merging to master in any case. 🙂

File: www/js/app.js
Comment (dheerajdlalwani) [reply www/js/app.js:394]: Thank you! I will test it.

Comment (Jaifroid) [reply www/js/app.js:394]: @dheerajdlalwani I also suggest you add a comment just above this block, so that devs know what this code is doing:

`// Empty and purge the article contents`

File: www/js/app.js
Comment (dheerajdlalwani) [reply www/js/app.js]: Oh alright. Sorry, I'll keep this in mind :)

**Comment by dheerajdlalwani** [comment]
I tested your suggestion, @Jaifroid and it works flawlessly! Thank you so much. Let me know if anything else is required.

Here, take a look!

image

**Review by Jaifroid — APPROVED** [review]

**Comment by Jaifroid** [comment]
@dheerajdlalwani Thank you very much, and also for testing thoroughly. It looks good now! I presume you don't see a button "Squash and merge" below the test results below? If you don't see it, or can't click it, let me know if you are ready to squash/merge, and I'll do it -- it will still appear on GitHub as your contribution/merge. NB it is important to squash down to a single commit, but GitHub can do this automatically.

**Comment by dheerajdlalwani** [comment]
Thank you so much, @Jaifroid. No, I don't see the Squash and Merge button. I think you will have to do it. And you don't have to thank me! It was absolutely my pleasure. I'm learning so much from this! Infact, I am enjoying it!

**Pull request merged**
